File: backend/app/Models/Trainer.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToTenant;
use Database\Factories\TrainerFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * user_id links this roster entry to the staff User account that logs
 * in as this trainer (Phase 26 — the Flutter Trainer App). Nullable —
 * see the migration docblock for why neither side requires the other.
 */
#[Fillable(['user_id', 'name', 'photo', 'specialty', 'specialties', 'experience', 'phone', 'email', 'bio', 'status', 'rating', 'sessions_completed', 'schedule'])]
class Trainer extends Model
{
    /** @use HasFactory<TrainerFactory> */
    use BelongsToTenant, HasFactory;

    protected function casts(): array
    {
        return [
            'specialties' => 'array',
            'schedule' => 'array',
            'rating' => 'decimal:2',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function members(): HasMany
    {
        return $this->hasMany(Member::class, 'trainer_id');
    }

    public function trainingPrograms(): HasMany
    {
        return $this->hasMany(TrainingProgram::class, 'trainer_id');
    }

    public function gymClasses(): HasMany
    {
        return $this->hasMany(GymClass::class, 'trainer_id');
    }

    public function personalTrainingSessions(): HasMany
    {
        return $this->hasMany(PersonalTrainingSession::class, 'trainer_id');
    }
}

File: mobile/lib/shared/models/trainer.dart
class Trainer {
  const Trainer({
    required this.id,
    this.userId,
    required this.name,
    this.photo,
    this.specialty,
    this.specialties = const [],
    this.experience,
    this.phone,
    this.email,
    this.bio,
    required this.status,
    required this.rating,
    this.sessionsCompleted,
  });

  factory Trainer.fromJson(Map<String, dynamic> json) => Trainer(
        id: json['id'] as int,
        userId: json['userId'] as int?,
        name: json['name'] as String? ?? '',
        photo: json['photo'] as String?,
        specialty: json['specialty'] as String?,
        specialties: (json['specialties'] as List?)?.map((e) => e.toString()).toList() ?? const [],
        experience: json['experience'] as String?,
        phone: json['phone'] as String?,
        email: json['email'] as String?,
        bio: json['bio'] as String?,
        status: json['status'] as String? ?? 'Active',
        rating: (json['rating'] as num?)?.toDouble() ?? 0,
        sessionsCompleted: (json['sessionsCompleted'] as num?)?.toInt(),
      );

  final int id;

  /// The staff User account this roster row is linked to, if any (Phase
  /// 26 — the Flutter Trainer App). Null for a trainer with no app login.
  final int? userId;
  final String name;
  final String? photo;
  final String? specialty;
  final List<String> specialties;
  final String? experience;
  final String? phone;
  final String? email;
  final String? bio;
  final String status;
  final double rating;
  final int? sessionsCompleted;
}
